src/modules/git_branch.rs

```diff
@@ -26,6 +26,11 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
 
     let repo = context.get_repo().ok()?;
 
+    let gix_repo = repo.open();
+    if config.ignore_bare_repo && gix_repo.is_bare() {
+        return None;
+    }
+
     if config.only_attached && repo.open().head().ok()?.is_detached() {
```

Comment:
Let's reuse the repo in the following line:
```suggestion
    if config.only_attached && gix_repo.head().ok()?.is_detached() {
```

Reply:
I've applied this changes